manager/state/raft/raft.go

```diff
@@ -1303,6 +1304,47 @@ func (n *Node) reportNewAddress(ctx context.Context, id uint64) error {
 	return n.transport.UpdatePeerAddr(id, newAddr)
 }
 
+// RaftMessageStream is the server endpoint for streaming Raft messages.
+// It accepts a stream of raft messages to be processed on this raft member,
+// returning a ProcessRaftMessageResponse when processing of the streamed messages is complete.
+// It is called from the Raft leader, which uses it to stream messages to this raft member.
+func (n *Node) RaftMessageStream(stream api.Raft_RaftMessageStreamServer) error {
+	var recvdMsg, assembledMessage *api.ProcessRaftMessageRequest
+	var err error
+	for {
+		recvdMsg, err = stream.Recv()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			log.G(stream.Context()).WithError(err).Error("error while reading from stream")
+			return err
+		}
+
```

Comment:
Validate that each message belongs to the same index and are to be concatenated.